src/ConnectedMode.UnitTests/Binding/ImportBeforeInstallTriggerTests.cs

```diff
@@ -34,56 +33,130 @@ public class ImportBeforeInstallTriggerTests
+        public void InvokeBindingChanged_ConnectedMode_ImportBeforeFileGeneratorIsCalled(SonarLintMode mode)
         {
-            var configProvider = SetUpConfigProvider(mode);
+            var activeSolutionTracker = CreateActiveSolutionBoundTrackerWihtBindingConfig(SonarLintMode.Standalone);
             var importBeforeFileGenerator = new Mock<IImportBeforeFileGenerator>();
 
-            var testSubject = CreateTestSubject(configProvider, importBeforeFileGenerator.Object);
-            testSubject.TriggerUpdate().Forget();
+            _ = CreateTestSubject(activeSolutionTracker.Object, importBeforeFileGenerator.Object);
+
+            importBeforeFileGenerator.Verify(x => x.WriteTargetsFileToDiskIfNotExists(), Times.Never);
 
+            var newBindingConfig = CreateBindingConfiguration(mode);
+            activeSolutionTracker.Setup(x => x.CurrentConfiguration).Returns(newBindingConfig);
+
+            activeSolutionTracker.Raise(x => x.PreSolutionBindingChanged += null, new ActiveSolutionBindingEventArgs(newBindingConfig));
             importBeforeFileGenerator.Verify(x => x.WriteTargetsFileToDiskIfNotExists(), Times.Once);
         }
 
-        private IConfigurationProvider SetUpConfigProvider(SonarLintMode mode)
+        [TestMethod]
+        public void InvokeBindingUpdated_ImportBeforeFileGeneratorIsCalled()
         {
-            var configProvider = new Mock<IConfigurationProvider>();
-            var bindingConfig = new BindingConfiguration(
-                                    new BoundSonarQubeProject(new Uri("http://localhost"), "test", ""),
-                                    mode, "");
+            var activeSolutionTracker = CreateActiveSolutionBoundTrackerWihtBindingConfig(SonarLintMode.Standalone);
+            var importBeforeFileGenerator = new Mock<IImportBeforeFileGenerator>();
+
+            _ = CreateTestSubject(activeSolutionTracker.Object, importBeforeFileGenerator.Object);
+
+            importBeforeFileGenerator.Verify(x => x.WriteTargetsFileToDiskIfNotExists(), Times.Never);
+
+            var newBindingConfig = CreateBindingConfiguration(SonarLintMode.Connected);
+            activeSolutionTracker.Setup(x => x.CurrentConfiguration).Returns(newBindingConfig);
+
+            activeSolutionTracker.Raise(x => x.PreSolutionBindingUpdated += null, EventArgs.Empty);
+            importBeforeFileGenerator.Verify(x => x.WriteTargetsFileToDiskIfNotExists(), Times.Once);
+        }
+
+        [TestMethod]
+        public void Dispose_UnsubscribesToEvent()
+        {
+            var activeSolutionTracker = new Mock<IActiveSolutionBoundTracker>();
 
-            configProvider.Setup(x => x.GetConfiguration()).Returns(bindingConfig);
+            var testSubject = CreateTestSubject(activeSolutionTracker.Object, Mock.Of<IImportBeforeFileGenerator>());
 
-            return configProvider.Object;
+            testSubject.Dispose();
```

Comment:
The product class isn't actually marked as disposable, but this test passes because `Dispose` is being called directly.

You could tweak the test to check that the class is disposable by casting explicitly to `IDisposable`:
```
  ((IDisposable)testSubject).Dispose();
```